File: optic/components/text/text.py
from pydantic import validate_arguments

import optic.components.text.text_pb2 as text_pb2
from optic.component_helpers import insert_component


@validate_arguments
def text(
  *,
  text: str,
  key: str | None = None,
):
  insert_component(
    key=key, type_name="text", proto=text_pb2.TextType(text=text)
  )

File: optic/components/text/text.ts
import { ChangeDetectionStrategy, Component, Input } from "@angular/core";
import {
  Key,
  Type,
} from "optic/optic/protos/ui_jspb_proto_pb/optic/protos/ui_pb";
import { TextType } from "optic/optic/components/text/text_jspb_proto_pb/optic/components/text/text_pb";

@Component({
  selector: "optic-text",
  templateUrl: "text.ng.html",
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TextComponent {
  @Input({ required: true }) type!: Type;
  @Input() key!: Key;
  _config!: TextType;

  ngOnChanges() {
    this._config = TextType.deserializeBinary(
      this.type.getValue() as Uint8Array,
    );
  }

  config(): TextType {
    return this._config;
  }
}
